modules/ROOT/pages/architecture/hexagonale_architecture.adoc

```diff
@@ -0,0 +1,203 @@
+It needs to be distinguished between inbound ports and outbound ports.
+
+| core.port.in
+| Inbound ports are the entry of the application. 
+They provide interfaces that are called from inbound adapters and hide the actual implementation. 
+A proposal of structuring inbound ports is naming them like single use cases (e.g. CancelReservationPort). 
+Each port should only provide a single method.
+.Design Decision
+[%collapsible]
+====
+Inbound Ports are not as relevant for the hexagonal architecture as the outbound ports. 
+Outbound ports are used for the dependency inversion pattern. 
+For inbound ports could also call the use cases directly. 
+Therefore, an pragmatic alternative would be leaving out the inbound ports.
+
+It was decided to include the inbound ports nonetheless. They should implement single use cases that are offered. 
+Each interface should clearly mark the use case that contains only one method.
+Use cases from the interface might be grouped logically in the use case implementation class.
+====
+
+| core.port.out
+| Outbound ports decouple the interaction with external systems. 
+This might include other services that are called, files that are written, databases, event streaming and everything the application is actively triggering outside of the core.
+Outbound ports should describe the business need for the communication (e.g. StoreReservationPort). How this is then realized depends on the adapter that implements it. 
+This way a technology can be easily replaced. 
+For example storing the reservation could be be realized in a first prototype by writing the objects to a file. 
+Later it could be replaced with a database.
+The core logic would be untouched by that.
+
+| [optional] core.service
+| Services can be considered as business helper classes. 
+They provide a reusable part of the applications business logic that is used by multiple use cases or that helps to structure the application in a logical way.
+Services are optional as they can be used, when there's a real need. Usually a use case should contain the business logic.
+
+| adapter
+a| Adapters connect the application core to the surrounding context. They have the following tasks:
+
+* Implement a specific protocol to connect to the context. E.g REST, JDBC, MQTT, ...
+* Maintain a data model that is necessary to communicate with the context
+* Translate the domain model from the core to that model or vice versa
+* Handle protocol specific errors
+* Log the interaction with the surrounding context
+
+| adapter.in
+| Inbound adapters specify connection points for everything that can trigger the business logic. 
+That might be interfaces (HTML, RPC, etc), Message Consumers or schedulers for batch processing.
+Inside the adapters further packages are differentiating the protocols that is used (e.g. `.rest`). 
```

Comment:
To align it with the discussion with Tim it would be better to name it `web`or `repository` for outbound adapters